### PR TITLE
Lesson 2

### PR DESCRIPTION
**Pull request by bwengals**

### CODE REVIEW

File: lesson_2.ipynb [hunk 1]
```diff
@@ -0,0 +1,2129 @@
+{
```
Comment (canyon289) [inline lesson_2.ipynb]: <p>Great point at how connecting the dots is not a good approach. Also if our method was to connect the dots how would we estimate uncertainty?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='108'/>

Comment (canyon289) [inline lesson_2.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #1.    ### HIDE THIS CODE
</pre><p>Great idea</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='109'/>

Comment (canyon289) [inline lesson_2.ipynb]: <p>Great idea plotting just a section of the data, focuses on just one part of the trend which is great and wont overwehlm people too much. You also picked a great part of the series to showcase these as the third point from the left is "out of band"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='113'/>

**Comment by canyon289** [comment]
I continue to be more excited every update I see

For my own sanity and convenience relinking our script here
https://github.com/IntuitiveBayes/gp_course_internal/blob/main/scripts/lesson_02.md

**Comment by bwengals** [comment]
@canyon289 Went ahead and pushed the latest updates to this branch.  Fixed the kernel equations, fixed the matplotib obj oriented stuff, and cleaned up mauna loa example some more.

**Comment by canyon289** [comment]
Nits on specific slides. Some of this is one me, im just listing it here
**Update**: @bwengals ignore this comment block as its all on Section 10 and 20, I realized this is likely because the hide code extension is not enabled on your computer. Please do read the next comment below for fixes on section 30

## Find a bigger mauna loa picture
This one looks fairly boring, would be nice to get a more tropical one

image

## Fix image being cut off
image

## Resize plot
image

## Fix PyMC Image
image

## Hide Plotting code 
image

## Hide Plotting code
image

## Fix slides here
image

## In section 10 and 20 fix up most slides
I'll stop commenting on each one

**Comment by canyon289** [comment]
Heres comments for Section 40. Please do fix these

## Bigger farmhouse picture
image

## Fix image size to fit into 1920x1080
image

## Rehide the code blocks
Please ensure the hide code block extension is enabled on your computer. Thes images are sized correctly, but if the extension is not there itll shift the image. See the comparison below

image
image

## Fix this slide
image

**Comment by canyon289** [comment]
In NB extensions please ensure you have
* Hide Input
* isort formatter
* Jupyterblack
* Split cells notebook

all enabled
image

**Comment by bwengals** [comment]
<p>good catch, lots of other plots missing these too </p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>Yup looks like it is, deleted this cell</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>thanks!</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>gotcha</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>fixed it up a bit</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>It was added back everywhere but here whoops.  I added a bit of text about the 2, too</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>I see what you're saying, but I think "short" and "long" is the typical nomenclature.  I think the intent of that is actually to conflate lengthscale with distance, like "short lengthscale" means "short distance effects".  There's nothing wrong with small, medium, large, but I'd resist standardizing one way or the other though since I think both are common "out there"</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>Yeah that's a cool example, I'll add that.  </p><p> </p><p>The lengthscale idea can be that general and more.  You "can" do things like a mahalanobis distance, where the lengthscales can have these sort of correlations between each other.  Never seen anyone do it.  </p><p> </p><p>I have seen where say x is very high dimensional, you can do a PCA like thing here too</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>good call, pretty much added this text</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>Looks good to me.  Added a bulled that we've defined kernel functions, what they mean, what they represent.</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bwengals** [comment]
<p>For sure.  I'm not sure about text/narrative here, since we're not building this up.  How do these sorts of demos go in the intro course?</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by canyon289** [comment]
Just a heads up
1. I added rise.css so youll need to repull this
2. Some of the slides metadata seems to be missing now, such as slide indicators, and hiding code cells. Those will need to be readded for things to work correctly, this can be done at the end though once all the content is in

**Comment by bwengals** [comment]
I guess I looked at 50 but forgot to leave comments.  I think it looks great.  I think we should merge this branch and then go from there

**Comment by canyon289** [comment]
Awesome, sounds like this entire lesson is "performance ready" then. Lets focus now then on that aspect in the coming week!

**Comment by bwengals** [comment]
should we merge this?  should rename the notebooks first though.  there are a couple others in here that are out of date

**Comment by canyon289** [comment]
Instead of merging, when were done we just copy  over the relevant files into the private repo and close this PR. Eventually were going to delete this whole repo, its just here temporarily so we can use reviewnb while in development

**Comment by canyon289** [comment]
To be clear Im happy to copy it over! Ill do it later today as well

**Comment by canyon289** [comment]
Superseded by https://github.com/IntuitiveBayes/GaussianProcessesAlpha

**Pull request closed without merging**